src/main/java/domain/GamePlayer.java

```diff
@@ -0,0 +1,76 @@
+package domain;
+
+import io.Message;
+import io.Printer;
+import io.Receiver;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class GamePlayer {
+    private static final int WINNER_CONDITION = 4;
+
+    private final Printer printer;
+    private final Receiver receiver;
+    private final Generator generator;
+
+    public GamePlayer() {
+        this.printer = new Printer();
+        this.receiver = new Receiver();
+        this.generator = new Generator();
+    }
+
+    void judgeAndMove(Car car, int randomNumber) {
+        if (randomNumber >= WINNER_CONDITION)
+            car.moveForward();
+    }
+
+    ArrayList<Car> inputNames() {
+        String messageCode = Message.GeneralMessages.INPUT_NAMEOFCAR.getMessage();
+        printer.printMessages(messageCode);
+        ArrayList<String> names = new ArrayList<>(Arrays.asList(receiver.receiveName()));
+
+        return makeCarList(names);
+    }
+
+    ArrayList<Car> makeCarList(List<String> names) {
+        ArrayList<Car> cars = new ArrayList<>();
+
+        for (int i = 0; i < names.size(); i++)
+            cars.add(new Car(names.get(i)));
+
+        return cars;
+    }
```

Comment:
```suggestion
    List<Car> inputNames() {
        String messageCode = Message.GeneralMessages.INPUT_NAMEOFCAR.getMessage();
        printer.printMessages(messageCode);
        List<String> names = new ArrayList<>(Arrays.asList(receiver.receiveName()));

        return makeCarList(names);
    }

    List<Car> makeCarList(List<String> names) {
        List<Car> cars = new ArrayList<>();

        for (int i = 0; i < names.size(); i++)
            cars.add(new Car(names.get(i)));

        return cars;
    }
```

자바의 유명한 규칙중엔 _program to interfaces not implementations_ 라는 부분이 있습니다.

[객체 지향 기초 이야기 3, 유연함(Flexibility)](https://javacan.tistory.com/entry/OO-Basic-3-Flexibility)

위 글 참고해보시면 좋을것 같아요 🙂